src/test/java/com/arpnetworking/metrics/common/integration/KafkaIT.java

```diff
@@ -0,0 +1,253 @@
+
+        // Observe records
+        final Observer observer = Mockito.mock(Observer.class);
+        _source.attach(observer);
+        _source.start();
+        for (ProducerRecord<Integer, String> expected : _producerRecords) {
+            Mockito.verify(observer, Mockito.timeout(TIMEOUT)).notify(_source, expected.value());
+        }
+    }
+
+    private String createTopicName() {
+        return "topic_" + UUID.randomUUID().toString();
+    }
+
+    private static void createTopic(final String topicName) {
+        try {
+            final Properties config = new Properties();
+            config.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA_SERVER);
+            config.setProperty(AdminClientConfig.CLIENT_ID_CONFIG, "adminClient");
+            final AdminClient adminClient = AdminClient.create(config);
+            final NewTopic newTopic = new NewTopic(topicName, 1, (short) 1);
+
+            // Create topic, which is async call
+            final CreateTopicsResult createTopicsResult = adminClient.createTopics(Collections.singleton(newTopic));
+
+            // Wait for it to complete
+            createTopicsResult.values().get(topicName).get();
+        } catch (final InterruptedException | ExecutionException e) {
+            if (!(e.getCause() instanceof TopicExistsException)) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+            // Ignore TopicExistsException
+        }
+    }
+
+    private static List<ProducerRecord<Integer, String>> createProducerRecords(final String topic, final int num) {
+        final List<ProducerRecord<Integer, String>> records = new ArrayList<>();
+        for (int i = 0; i < num; i++) {
+            records.add(new ProducerRecord<>(topic, i, "value" + i));
+        }
+        return records;
+    }
+
+    private void sendRecords(final List<ProducerRecord<Integer, String>> records) {
+        final Map<String, Object> producerProps;
+        producerProps = Maps.newHashMap();
+        producerProps.put(ProducerConfig.ACKS_CONFIG, "all");
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA_SERVER);
+        producerProps.put(ProducerConfig.CLIENT_ID_CONFIG, "producer0");
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, KEY_SERIALIZER);
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, VALUE_SERIALIZER);
+        producerProps.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, "transactional-id");
+
+        final KafkaProducer<Integer, String> producer = new KafkaProducer<>(producerProps);
+        producer.initTransactions();
+        try {
+            producer.beginTransaction();
+            for (ProducerRecord<Integer, String> record : records) {
+                producer.send(record);
+            }
+            producer.commitTransaction();
+        } catch (final ProducerFencedException | OutOfOrderSequenceException | AuthorizationException e) {
+            producer.close();
```

Comment:
It might be appropriate to `Assert.fail()` here? If Kafka gets messed up somehow and doesn't accept the publication, failing here would make the cause of the failure immediately obvious, rather than causing downstream test failures.

(same goes for the next `catch` branch too)